independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/ClassLoadingResource.java

```diff
@@ -5,13 +5,22 @@
 
 public interface ClassLoadingResource {
 
+    /**
+     * A lifecycle hook that should be called when the ClassLoader to which this resource belongs to
+     * is constructed
+     */
+    void init(ClassLoader runnerClassLoader);
```

Comment:
just wondering - do we really want to impose that a ClassLoadingResource belongs to a unique classloader ever?

Reply:
We indeed might want to revisit this in the future, but for now we only have single `RunnerClassLoader` so it's the YAGNIest way to remove the slow map :)